hw/ip/keymgr/doc/_index.md

```diff
@@ -0,0 +1,211 @@
+
+This is the second operational state of the key manager.
+This state is reached through another invocation of the KMAC operation using the previous working state as the key, and other inputs as data.
+The output of the KMAC operation replaces the previous value of the working state, and the new value becomes the `OwnerIntermediateKey`.
+
+The relevant data inputs are:
+*  `OwnerRootSecret`: Secret seed from flash.
+*  `SoftwareBinding`: A software programmed value representing the first owner code to be run.
+
+Once the `OwnerIntermediateKey` is created, software can request key manager to advance state, generate output key or generate output identity.
+The key used for all 3 functions is the `OwnerIntermediateKey`.
+
+The advancement from this state to the next is irreversible during the current power cycle.
+
+### OwnerRootKey
+
+This is the last operational state of the key manager.
+This state is reached through another invocation of the KMAC operation using the previous working state as the key, and other inputs as data.
+The output of the KMAC operation replaces the previous value of the working state, and the new value becomes the `OwnerRootKey`.
+
+The relevant inputs are:
+*   `SoftwareBinding` - A software programmed value representing the owner kernel code.
+
+Once the `OwnerRootKey` is created, software can request key manager to advance state, generate output key or generate output identity.
+An advance command invoked from `OwnerRootKey` state simply moves the state to `Disabled`.
+
+The generate output and generate identity functions use `OwnerRootKey` as the KMAC key.
+The advancement from this state to the next is irreversible during the current power cycle.
+
+### Disabled
+
+`Disabled` is a terminal state where the key manager is no longer operational.
+When entering Disabled random values are used to compute a new random value.
+
+## Life Cycle Connection
+The function of the key manager is directly tied to the life cycle controller.
+During specific life cycle states, the key manager is explicitly disabled.
+
+When disabled, key manager will not any advance or generate commands and if currently in an operational state, will automatically transition to `Disabled`.
+
+
+## Commands in Each State
+During each state, there are 3 valid commands software can issue:
+*  Advance state
+*  Output generation
+*  Identity generation
+
+The software is able to select a command and trigger the key manager FSM to process one of the commands.
+If a command is valid during the current working state, it is processed and acknowledged when complete.
+If a command is invalid, the key manager FSM processes with random, dummy data, but does not update working state or relevant output registers.
+
+For each valid command, a set of inputs are selected and sequenced to the KMAC module.
+
+## Generating Output Key
+The generate output command is composed of 2 options
+*  Generate output key for software
+*  Generate output key for hardware
```

Comment:
maybe put "(generate-output-sw)" and "(generate-output-hw)" after these two bullets if that's how you'll refer to them later?

Reply:
thanks, yes this is better.